FlowCal/excel_ui.py

```diff
@@ -1054,18 +1083,44 @@ def add_samples_stats(samples_table, samples):
                                         channel + ' CV',
                                         FlowCal.stats.cv(sample, channel))
                 samples_table.set_value(row_id,
-                                        channel + ' Geom. Std',
-                                        FlowCal.stats.gstd(sample, channel))
-                samples_table.set_value(row_id,
-                                        channel + ' Geom. CV',
-                                        FlowCal.stats.gcv(sample, channel))
-                samples_table.set_value(row_id,
                                         channel + ' IQR',
                                         FlowCal.stats.iqr(sample, channel))
                 samples_table.set_value(row_id,
                                         channel + ' RCV',
                                         FlowCal.stats.rcv(sample, channel))
 
+                # For geometric statistics, first check for non-positive events.
+                # If found, throw a warning and calculate statistics on positive
+                # events only.
+                if np.any(sample[:, channel] <= 0):
+                    # Separate positive events
+                    sample_positive = sample[sample[:, channel] > 0]
+                    # Throw warning
+                    msg = "Geometric statistics for channel" + \
+                        " {} calculated on positive events".format(channel) + \
+                        " only ({:.1f}%). ".format(
+                            100.*sample_positive.shape[0]/sample.shape[0])
+                    warnings.warn("On sample {}: {}".format(row_id, msg))
+                    # Write warning message to table
+                    if samples_table.loc[row_id, 'Analysis Notes']:
+                        msg = samples_table.loc[row_id, 'Analysis Notes'] + msg
+                    samples_table.set_value(row_id, 'Analysis Notes', msg)
+                else:
+                    sample_positive = sample
+                # Calculate and write geometric statistics
+                samples_table.set_value(
+                    row_id,
+                    channel + ' Geom. Mean',
+                    FlowCal.stats.gmean(sample_positive, channel))
+                samples_table.set_value(
+                    row_id,
+                    channel + ' Geom. Std',
+                    FlowCal.stats.gstd(sample_positive, channel))
+                samples_table.set_value(
+                    row_id,
+                    channel + ' Geom. CV',
+                    FlowCal.stats.gcv(sample_positive, channel))
```

Comment:
does processing the geometric statistics at the end of `add_samples_stats()` change the order in which the statistics are reported in the XLSX sheet? I wasn't particularly tied to a previous order, but just wanted to check if this was done on purpose.